### PR TITLE
Feature/galera autofix

### PR DESCRIPTION
**Pull request by julsemaan**
# Description
Service to automatically fix broken Galera clusters

# Impacts
Galera cluster stack in PF

# Delete branch after merge
YES

# NEWS file entries
## New Features
* Added service to correct broken Galera cluster members

# UPGRADE file entries

This release adds a new service that will automatically attempt to recover broken Galera cluster members and can also perform a full recovery of a Galera cluster. These automated decisions may lead to potential data loss. If this is not acceptable for you disable the galera-autofix service in pf.conf or in "System Configuration->Services". More details and documentation is available in the "The galera-autofix service" section of the clustering guide.

### CODE REVIEW

File: go/galeraautofix/main.go [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+	log.LoggerWContext(ctx).Info(fmt.Sprintf("Waiting %s before we start checking for DB issues", startWait))
+	time.Sleep(startWait)
+	log.LoggerWContext(ctx).Info("Activating galera-autofix")
+
+	decisionLoop(ctx)
+}
+
+func getSeqnoReport(ctx context.Context, nodes *NodeList) bool {
+	log.LoggerWContext(ctx).Info("Started server to listen to sequence numbers from peers")
+	sAddr, err := net.ResolveUDPAddr("udp", ":4253")
```
Comment (fdurand) [inline go/galeraautofix/main.go]: make 4253 a constant

File: lib/pf/services/manager/galera_autofix.pm [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+=head1 AUTHOR
+
+Inverse inc. <info@inverse.ca>
+
+
+
+=head1 COPYRIGHT
+
+Copyright (C) 2005-2019 Inverse inc.
```
Comment (fdurand) [inline lib/pf/services/manager/galera_autofix.pm]: 2020

**Review by fdurand — COMMENTED** [review]
Looks good, juste need to be tested.
Also it looks that it miss the port 4253 in iptables.

**Comment by julsemaan** [comment]
Adjusted based on comments + rebased

File: conf/iptables.conf.example [hunk 3]
```diff
@@ -141,6 +141,8 @@
 :input-highavailability-if - [0:0]
 #SSH
 -A input-highavailability-if --match state --state NEW --match tcp --protocol tcp --dport 22 --jump ACCEPT
+#Galera autofix
+-A input-highavailability-if --protocol tcp --match tcp --dport 4253 --jump ACCEPT
```
Comment (fdurand) [inline conf/iptables.conf.example]: good try but it's udp

**Pull request merged**